File: CreaObjetos/CreaObjetos/GeneraCodigo/generaForma.vb
Public Class generaForma
    Public neimespeis As String
    Public clace As Clase

    Public Function generaForma() As String
        Dim code As New String("")
        code &= "Imports Sist" & neimespeis & "." & neimespeis & vbCrLf
        code &= "Imports Sist" & neimespeis & ".Maneja" & neimespeis & vbCrLf
        code &= " Public Class frm" & Me.clace.NombreTabla & vbCrLf
        code &= "Public mnj" & Me.clace.nombre.Substring(0, 3) & " as new maneja" & Me.clace.NombreTabla & vbCrLf
        code &= Me.generaLoad() & vbCrLf
        code &= "End Class" & vbCrLf
        Return code

    End Function


    Public Function generaLoad() As String
        Dim code As New String("")
        code &= "Private Sub Frm" & Me.clace.NombreTabla
        code &= "_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) "
        code &= " Handles MyBase.Load" & vbCrLf
        code &= "Me.mnj" & Me.clace.nombre.Substring(0, 3) & ".ctlBusq = Me.CtlbuscaReg" & vbCrLf
        code &= "Me.mnj" & Me.clace.nombre.Substring(0, 3) & ".ctlMaestro = Me.CtlABCMaestro" & vbCrLf
        code &= "Me.CtlbuscaReg = Me.mnj" & Me.clace.nombre.Substring(0, 3) & ".ctlBusq" & vbCrLf
        code &= "Me.mnj" & Me.clace.nombre.Substring(0, 3) & ".cargaControles(Me.pnlDatos.Controls)" & vbCrLf
        code &= "Me.CtlbuscaReg.mnjObj = Me.mnj" & Me.clace.nombre.Substring(0, 3) & vbCrLf
        code &= "mnj" & Me.clace.nombre.Substring(0, 3) & ".iniciar()" & vbCrLf
        code &= "End Sub " & vbCrLf


   
        Return code

    End Function






End Class

File: trunk/CreaObjetos/CreaObjetos.ConvertedToC#/GeneraCodigo/generaForma.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
namespace WindowsApplication1
{
	public class generaForma
	{
		public string neimespeis;

		public Clase clace;
		public string creaForma()
		{
			string code = string.Empty;
			code += "Imports Sist" + neimespeis + "." + neimespeis + Constants.vbCrLf;
			code += "Imports Sist" + neimespeis + ".Maneja" + neimespeis + Constants.vbCrLf;
			code += " Public Class frm" + this.clace.NombreTabla + Constants.vbCrLf;
			code += "Public mnj" + this.clace.nombre().Substring(0, 3) + " as new maneja" + this.clace.NombreTabla + Constants.vbCrLf;
			code += this.generaLoad() + Constants.vbCrLf;
			code += "End Class" + Constants.vbCrLf;
			return code;

		}


		public string generaLoad()
		{
			string code = string.Empty;
			code += "Private Sub Frm" + this.clace.NombreTabla;
			code += "_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) ";
			code += " Handles MyBase.Load" + Constants.vbCrLf;
			code += "Me.mnj" + this.clace.nombre().Substring(0, 3) + ".ctlBusq = Me.CtlbuscaReg" + Constants.vbCrLf;
			code += "Me.mnj" + this.clace.nombre().Substring(0, 3) + ".ctlMaestro = Me.CtlABCMaestro" + Constants.vbCrLf;
			code += "Me.CtlbuscaReg = Me.mnj" + this.clace.nombre().Substring(0, 3) + ".ctlBusq" + Constants.vbCrLf;
			code += "Me.mnj" + this.clace.nombre().Substring(0, 3) + ".cargaControles(Me.pnlDatos.Controls)" + Constants.vbCrLf;
			code += "Me.CtlbuscaReg.mnjObj = Me.mnj" + this.clace.nombre().Substring(0, 3) + Constants.vbCrLf;
			code += "mnj" + this.clace.nombre().Substring(0, 3) + ".iniciar()" + Constants.vbCrLf;
			code += "End Sub " + Constants.vbCrLf;



			return code;

		}






	}
}
